homeassistant/components/switch/raincloud.py

```diff
@@ -0,0 +1,119 @@
+"""
+Support for Melnor RainCloud sprinkler water timer.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.raincloud/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.raincloud import (
+    ALLOWED_WATERING_TIME, CONF_ATTRIBUTION, CONF_WATERING_TIME,
+    DATA_RAINCLOUD, DEFAULT_WATERING_TIME, RainCloudHub)
+from homeassistant.components.switch import SwitchDevice, PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_MONITORED_CONDITIONS, ATTR_ATTRIBUTION)
+
+DEPENDENCIES = ['raincloud']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Sensor types: label, desc, unit, icon
+SENSOR_TYPES = {
+    'auto_watering': ['Automatic Watering', 'mdi:autorenew'],
+    'manual_watering': ['Manual Watering', 'water-pump'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SENSOR_TYPES)):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_WATERING_TIME, default=DEFAULT_WATERING_TIME):
+        vol.All(vol.In(ALLOWED_WATERING_TIME)),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up a sensor for a raincloud device."""
+    raincloud_hub = hass.data[DATA_RAINCLOUD]
+    raincloud = raincloud_hub.data
+    default_watering_timer = config.get(CONF_WATERING_TIME)
+
+    sensors = []
+    for sensor_type in config.get(CONF_MONITORED_CONDITIONS):
+        # create an sensor for each zone managed by faucet
+        for zone in raincloud.controller.faucet.zones:
+            sensors.append(
+                RainCloudSwitch(hass,
+                                zone,
+                                sensor_type,
+                                default_watering_timer))
+
+    add_devices(sensors, True)
+    return True
+
+
+class RainCloudSwitch(RainCloudHub, SwitchDevice):
+    """A switch implementation for raincloud device."""
+
+    def __init__(self, hass, data, sensor_type, default_watering_timer):
```

Comment:
If you move the init to the intermediate class, you overwrite the init in this class since it has a special case with the watering timer.

```py
def __init__(self, default_watering_timer, *args):
```